fclient/crosssigning.go

```diff
@@ -46,6 +47,49 @@ type CrossSigningKey struct {
 
 func (s *CrossSigningKey) isCrossSigningBody() {} // implements CrossSigningBody
 
+func (s *CrossSigningKey) Equal(other *CrossSigningKey) bool {
```

Comment:
It would be nice to document what the comparison is taking into account, i.e. [there can be different equality implementations](https://github.com/matrix-org/matrix-rust-sdk/blob/c50358366fe7f25606cb5494071391c39f74f43a/crates/matrix-sdk-crypto/src/types/cross_signing/mod.rs#L35-L45) depending on the use-case. I presume here we care about the fact that a new object needs to be stored in the DB and we test for equality of all fields.

Reply:
Exactly. The `Equal` method here checks all fields and receives the `MasterKey`, `SelfSigningKeys` and `UserSigningKeys` from Dendrite, see [this](https://github.com/element-hq/dendrite/pull/3471/files#diff-a9dcd4fed4827bd67f59b29c2ee48f7e819d5f97c20d7d91423e4f7f6d8cdb36R136-R150) (which should be much like https://github.com/element-hq/synapse/pull/16943